cmd/contact-auditor/main.go

```diff
@@ -0,0 +1,203 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/cmd"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/policy"
+)
+
+var queryInterrupted bool
+
+type contactAuditor struct {
+	db     *sql.DB
+	logger blog.Logger
+	clk    clock.Clock
+	grace  time.Duration
```

Comment:
nit: I like to always arrange "important" fields (i.e. things that are functional, like the grace period and the database connection) before "infrastructure" fields (i.e. the logger and the clock that is only used for testing). It's definitely not important, but it helps the code be self-documenting. Speaking of which: you could document the `grace` field here, so a user knows exactly what kind of thing the grace period is applied to.